server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -151,11 +151,11 @@ func (r *Runner) Plan(ctx workflow.Context, root *terraform.LocalRoot, serverURL
 	return response, nil
 }
 
-func (r *Runner) waitForPlanReview(ctx workflow.Context, root *terraform.LocalRoot) PlanStatus {
+func (r *Runner) waitForPlanReview(ctx workflow.Context, root *terraform.LocalRoot, planSummary terraform.PlanSummary) PlanStatus {
 	// Wait for plan review signal
 	var planReview PlanReviewSignalRequest
 
-	if root.Root.Plan.Approval == terraform.AutoApproval {
+	if root.Root.Plan.Approval == terraform.AutoApproval || planSummary.IsEmpty() {
```

Comment:
do you think it makes sense to move this logic in the calling function itself rather than in `waitForPlanReview` since it doesn't technically relate to waiting for plan review?